src/puppetlabs/puppetdb/helpers.clj

```diff
@@ -0,0 +1,89 @@
+(ns puppetlabs.puppetdb.helpers
+  (:require [clojure.java.io :as io]
+            [clojure.tools.logging :as log]
+            [compojure.core :as compojure]
+            [puppetlabs.kitchensink.core :as kitchensink]
+            [puppetlabs.puppetdb.archive :as archive]
+            [puppetlabs.puppetdb.cheshire :as json]
+            [puppetlabs.puppetdb.cli.export :as export]
+            [puppetlabs.puppetdb.cli.import :as import]
+            [puppetlabs.puppetdb.http :as http]
+            [puppetlabs.puppetdb.middleware :as mid]
+            [puppetlabs.trapperkeeper.core :as trapperkeeper]
+            [ring.middleware.multipart-params :as mp]
+            [ring.util.io :as rio]))
+
+(def api-version :v4)
+
+(defn import-app
+  [infile command-fn]
+  (import/import! infile (partial import/process-tar-entry
+                                  command-fn)))
+
+(defn completion-wrapper
```

Comment:
This function (as @mullr would say) kind of has a "code-smell", it's basically a translation from an `href` in our entities to an entity/query for our in-process querying.